src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/stønadsperiode/StønadsperiodeService.kt

```diff
@@ -1,38 +1,105 @@
 package no.nav.tilleggsstonader.sak.vilkår.stønadsperiode
 
+import no.nav.tilleggsstonader.sak.behandling.BehandlingService
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.feilHvis
 import no.nav.tilleggsstonader.sak.vilkår.stønadsperiode.domain.Stønadsperiode
 import no.nav.tilleggsstonader.sak.vilkår.stønadsperiode.domain.StønadsperiodeRepository
 import no.nav.tilleggsstonader.sak.vilkår.stønadsperiode.dto.StønadsperiodeDto
 import no.nav.tilleggsstonader.sak.vilkår.stønadsperiode.dto.tilSortertDto
 import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.VilkårperiodeService
 import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
 import java.util.UUID
 
 @Service
 class StønadsperiodeService(
+    private val behandlingService: BehandlingService,
     private val stønadsperiodeRepository: StønadsperiodeRepository,
     private val vilkårperiodeService: VilkårperiodeService,
 ) {
     fun hentStønadsperioder(behandlingId: UUID): List<StønadsperiodeDto> {
         return stønadsperiodeRepository.findAllByBehandlingId(behandlingId).tilSortertDto()
     }
 
+    @Transactional
     fun lagreStønadsperioder(behandlingId: UUID, stønadsperioder: List<StønadsperiodeDto>): List<StønadsperiodeDto> {
-        // TODO valider behandling er ikke låst
+        feilHvis(behandlingService.hentBehandling(behandlingId).status.behandlingErLåstForVidereRedigering()) {
+            "Kan ikke lagre stønadsperioder når behandlingen er låst"
+        }
         validerStønadsperioder(behandlingId, stønadsperioder)
-        stønadsperiodeRepository.deleteAll() // TODO skal vi finne og oppdatere de som har blitt oppdatert/slettet
-        return stønadsperiodeRepository.insertAll(
-            stønadsperioder.map {
-                Stønadsperiode(
-                    id = it.id ?: UUID.randomUUID(),
-                    behandlingId = behandlingId,
+        return lagreOgOppdaterStønadsperioder(behandlingId, stønadsperioder)
```

Comment:
Syntes navngivningen på flere av funksjonene her ble litt like. Kunne man heller hatt at en funksjon har alle stegene og kaller metoder hver for seg, enn at hver funksjon kaller videre på neste?

F.eks. 
1. Valider
2. Finn tidligere
3. Slett de som skal slettes
4. Oppdater de som skal oppdateres
5. Nye? (litt usikker på hva som skjer der)
6. Lagre ned

(Skrev denne før jeg har gått gjennom alt så ikke sikkert stegene er helt riktig, men overordnet flyt og navngivning ble litt kronglete nå)

Reply:
Det var egentlige for å dele opp validering og det som faktiskt oppdaterer perioder.
I tillegg så var det litt felles-behov for oppdatering/endring, som ikke sletting hadde

Har inlinet den https://github.com/navikt/tilleggsstonader-sak/pull/163/commits/3ca76b5bd66effc8b17c39ea32b3f6b37d820ff4

Reply:
Syntes det ble mer ryddig tror jeg! Forstod hvertfall mer nå. 

Kunne det vært noe at man startet med å finne de ulike delene? Typ `finnSlettedePerioder()`, `finnOppdatertePerioder()`, `finnNyePerioder()`?

Reply:
Føler ikke noe behov for en egen metode for det, det gjøres på en måte allerede inne i de ulike metodene?